capstone/src/components/User/Adventurer/ExperienceInfo/ExperienceInfo.jsx

```diff
@@ -18,23 +18,30 @@ const API_KEY = process.env.REACT_APP_GOOGLE_MAPS_API_KEY;
 const baseURL = process.env.REACT_APP_BASE_URL;
 
 async function submitRate(username, experienceId, reviews) {
-  const values = { username, experienceId, reviews };
-  return axios.post(`${baseURL}/review/rate`, values);
+  return axios.post(
+    `${baseURL}/adventure/rate`,
+    { reviews },
+    { headers: { username, experienceId } },
+  );
 }
 
 function getPhotoReference(restaurant) {
   return restaurant.photos ? restaurant.photos[0].photo_reference : null;
 }
 
-export default function ExperienceInfo({ onSubmit }) {
+export default function ExperienceInfo({ onUpdateRestaurants }) {
   const { username } = useContext(UserContext);
   const { restaurants } = useContext(AdventurerContext);
+  // All feedback information (including distance, which is not rateable)
   const feedbackInfo = useContext(FeedbackContext);
   const [indexRestaurant, setIndexRestaurant] = useState(0);
   const [newReview, setNewReview] = useState([]);
   // Get the restaurant viewing now
   const currentRestaurant = restaurants.length > 0 ? restaurants[indexRestaurant] : null;
-  const isRated = currentRestaurant?.review != null;
+  // QUESTION: Why is it not responding to changes (derived state)
+  // currentRestaurant.review not updated by setter. Queued. Till when will it be triggered?
+  // Thinking about not a derived state but another independent state for manipulation
+  const isRated = currentRestaurant ? currentRestaurant.review != null : false;
```

Comment:
The component will be re-rendered when 1) its props change or 2) its state (via useState) change.

Your change looks correct. I'd expect the component to rerender/update when indexRestaurant changes. Does that not happen?

Reply:
I think the issue is on line 66-70 (Githhub won't let me comment on it!)

We're checking `isRated` but it's not in the dependencies array.